scripts/ci/gitlab/pipeline/publish.yml

```diff
@@ -233,11 +233,19 @@ publish-draft-release:
 
 publish-crates:
   extends:                         .publish-crates-template
+  # publish-crates should only be run if publish-crates-locally passes
+  needs:
+    - job:                         publish-crates-locally
+      artifacts:                   false
 
 publish-crates-manual:
   extends:                         .publish-crates-template
   when:                            manual
   interruptible:                   false
+  # publish-crates should only be run if publish-crates-locally passes
+  needs:
+    - job:                         publish-crates-locally
+      artifacts:                   false
 
 publish-crates-locally:
```

Comment:
Could you rename this job to `check-crate-publishing`? It'll be easier to understand that it's a dependency of `publish-crates` that way